packages/hierarchies/src/hierarchies/HierarchyFiltering.ts

```diff
@@ -3,42 +3,50 @@
  * See LICENSE.md in the project root for license terms and full copyright notice.
  *--------------------------------------------------------------------------------------------*/
 
-import { HierarchyNode, NonGroupingHierarchyNode } from "./HierarchyNode.js";
+import { Props } from "@itwin/presentation-shared";
+import { HierarchyNode, NonGroupingHierarchyNode, ParentHierarchyNode } from "./HierarchyNode.js";
 import { HierarchyNodeIdentifier, HierarchyNodeIdentifiersPath } from "./HierarchyNodeIdentifier.js";
 import { GenericNodeKey, HierarchyNodeKey, InstancesNodeKey } from "./HierarchyNodeKey.js";
 
 /** @public */
 export interface FilteringPathAutoExpandDepthInPath {
   /**
-   * Depth that tells which nodes in the filtering path should be expanded.
-   *
    * Use when you want to expand up to specific instance node and don't care about grouping nodes.
    *
-   * **NOTE**: All nodes that are up to `depthInPath` will be expanded. Node at the `depthInPath` position won't be expanded.
+   * **Use case example:**
+   *
+   * All nodes up to `Element1` should be expanded in the following hierarchy:
+   * - Node1
+   *   - GroupingNode1
+   *     - GroupingNode2
+   *       - Element1
+   *       - Element2
+   * Then provide ` { path: [Node1, Element1], autoExpand: { depthInPath: 2 } }`
+   *
+   * **NOTE**: All nodes that are up to `depthInPath` will be expanded **except** filter targets.
    */
   depthInPath: number;
 }
 
 /** @public */
 export interface FilteringPathAutoExpandDepthInHierarchy {
   /**
-   * Depth that tells which nodes in the hierarchy should be expanded.
-   *
    * This should take into account the number of grouping nodes in hierarchy.
```

Comment:
I think docs for this attribute should start with a one-sentence description of what it's for.